docs/java/features/odata/type-safe-client-odata-v4.mdx

```diff
@@ -211,45 +211,35 @@ The described API is subject to change
 #### Execute Batch Request
 The `service` object offers the method `batch` which allows access to the batch-related methods that help build the batch request.
 
-Multiple single requests (like `createBusinessPartnerAddress`) can be wrapped into so-called change sets.
-Each change set is opened with `beginChangeSet` and closed with `endChangeSet`. You can wrap multiple change sets into one batch request.
+Multiple single requests (like `createBusinessPartner`) can be wrapped into so-called change sets.
+A change set can be added by `addChangeset` and the corresponding create, update or delete operations can be passed as partmenters into it. You can wrap multiple change sets into one batch request.
+Similarly, retrieve operations can be added to the batch request by using `addReadOperations`.
 
-Use `executeRequest` to issue the batch request to the remote system. We receive an instance of `BatchResponse` as result object.
+Use `execute` to issue the batch request to the remote system. We receive an instance of `BatchResponse` as result object.
 
 ```java
 final BatchResponse result =
     service
         .batch()
-        .beginChangeSet()
-        .createBusinessPartnerAddress(address1)
-        .createBusinessPartnerAddress(address2)
-        .endChangeSet()
-        .executeRequest(destination);
+        .addReadOperations(businessPartners, businessPartnerByKey)
+        .addChangeset(createBusinessPartner1(), createBusinessPartner2(), deleteBusinessPartner())
+        .execute(destination);
 ```
 
 #### Access Batch Response
-Use the method `get` to access the individual change set responses by index.
+Use `getModificationResult` to access response for a specific create, update or delete operation.
+
 ```java
-final Try<BatchResponseChangeSet> changeSetTry = result.get(0);
+final ModificationResponse<BusinessPartner> createResult = response.getModificationResult(createBusinessPartner1);
+final ModificationResponse<BusinessPartner> deleteResult = response.getModificationResult(deleteBusinessPartner);
 ```
 
-The returned `Try` allows checking if the respective change set was processed successfully by the remote server.
-As per the specification, either all operations or none within the same change set are successful. Hence, if `changeSetTry.isSuccess()` is true,
-you can derive that all operations in the change set are succcessful.
-
-In case the change set contained at least one create operation, you can gain access to the created entity representations by using the
-method `getCreatedEntities`.
+Similarly, use the method `getReadResult` to access response for a specific retrieve request.
```

Comment:
For change results we should point out how to obtain the result of a change set. If the change set was a failure then evaluating the result of any of the contained requests will contain the failure of the change set.

Reply:
Done